### DISCUSSION TITLE
Use of execution_date for partial day processing

### DISCUSSION THREAD

**Discussion by mumblemaker** [apache/airflow#23208] [Q&A] [answered]
Hi all. Hope someone can help with the below.

The concept of the execution_date has been asked and well explained - the execution date is the left-most date/time of a given period. However what I'm unable to find is the best practice for using the execution_date (through the ds_macros) when the range of time for each "date" does not fit within a 24-hour window.

To give an example, we collect data between 9am and 5pm each day (i.e. business hours) and wish to run a DAG to process that days data at 17:15. However, with an cron-like schedule of 17:15 each day, Airflow assume that a 24-hour period and the execution_date would be for the previous day.

To give a concrete example: the run we'd like to do for 25th April at 17:15 would assume that the reporting period would start from 24th April 17:15 (we'd expect it to be perhaps 25th April 09:00).

One way around this is to use a macro to increase the execution_date by one day. 
`{{ macros.ds_add(ds, 1)}}`
However, the execution_date would still be the previous day which could lead to some confusion (let me know if there's a flaw in this thinking).

Another option is to trigger at midnight so that the execution_date would be the date we would expect, however this would mean the DAG doesn't run until the end of the  day when we'd like to have the results available shortly after the end of the business day.

Could someone give some insight or best practices for handling these cases?

**Comment by potiuk** [comment, marked as answer]
In Airflow 2.2.0 you do not have execution_date - you have logical date and the whole concept about scheduling around the data intervals have been changed. In particular run_id is the primary key of the run and it can be anything., 

Please take a look there - at the https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-39+Richer+scheduler_interval and you find that it will be easier to fit your data ranges there.

**Reply by mumblemaker** [reply]
That document is very helpful. Thank you!
